src/shaders/disk_andrews2009_frag.glsl

```diff
@@ -47,14 +47,11 @@ void main(void)
 {
     // Compute the radius and height of this fragment
     float radius = sqrt(ModelPosition.x * ModelPosition.x + ModelPosition.y * ModelPosition.y);
-    float height = abs(ModelPosition.z);
+    float height = atan(abs(ModelPosition.z),radius);
```

Comment:
I am using tan2 because if you look at the formalism for Andrews, it defines the height as pi/2 - theta where theta = 0 is the midplane. Essentially height is only equal to z for small angles which is (kinda) the case for Pascucci.  I checked both definitions with Simtoi and saw no obvious problems. The tan2 formalism is simply more rigorous as it does not rely on the small angle approximation.

Peace,
Rob

Sent from my iPhone

> On Jul 1, 2015, at 6:43 PM, Brian Kloppenborg <notifications@github.com> wrote:
> 
> In src/shaders/disk_andrews2009_frag.glsl:
> 
> > @@ -47,14 +47,11 @@ void main(void)
> >  {
> >      // Compute the radius and height of this fragment
> >      float radius = sqrt(ModelPosition.x * ModelPosition.x + ModelPosition.y * ModelPosition.y);
> > -    float height = abs(ModelPosition.z);
> > +    float height = atan(abs(ModelPosition.z),radius);
> Why are you using atan2 here? The z value is the height above the midplane which is the height by definition.
> 
> —
> Reply to this email directly or view it on GitHub.
>